geode-core/src/main/java/org/apache/geode/cache/wan/GatewaySender.java

```diff
@@ -414,5 +415,12 @@
    */
   void destroy();
 
-
+  /**
+   *
+   * @return {@code true} if this instance is an asynchronous-event-queue, {@code false} if is a
+   *         gateway-sender.
+   */
+  default boolean isAsynchronousEventQueue() {
+    return getId().contains(AsyncEventQueueImpl.ASYNC_EVENT_QUEUE_PREFIX);
+  }
```

Comment:
I'm not sure that this method should be on the GatewaySender public API. Maybe we need an InternalGatewaySender interface like some of the other entities?

Reply:
Hello @boglesby,
I'll revert the changes to `GatewaySender` and leave the logic where it was before. Adding an internal interface sounds good, but I think it's worth opening another JIRA and respective pull request instead of adding the changes within this one.
Cheers.